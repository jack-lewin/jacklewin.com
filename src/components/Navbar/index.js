import React, { Fragment } from 'react';
import { Link } from 'gatsby';
import navbarProps, { defaultProps } from '../../prop-types/components/navbar';
import styles from './styles.module.scss';

const Navbar = ({ breadcrumbs }) => (
  <nav className={styles.container} role='navigation' aria-label='main-navigation'>
    <Link to='/'>Jack Lewin</Link>

    {breadcrumbs.map(({ path, text }) => (
      <Fragment>
        <span className={styles.divider}>/</span>
        <Link to={path}>{text}</Link>
      </Fragment>
    ))}
  </nav>
);

Navbar.propTypes = navbarProps;

Navbar.defaultProps = defaultProps;

export default Navbar;
