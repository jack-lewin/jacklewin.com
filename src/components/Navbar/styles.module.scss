@import '../../functions.scss';

.container {
  padding: spacing(small) 0;
}

.divider {
  opacity: 0.2;
  padding: 0 spacing(small);
}
