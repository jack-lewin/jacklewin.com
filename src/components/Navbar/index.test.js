import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'gatsby';
import Navbar from '.';

describe('Navbar', () => {
  const baseProps = {
    breadcrumbs: [],
  };

  it('should render as expected', () => {
    const element = shallow(<Navbar {...baseProps} />);
    expect(element).toMatchSnapshot();
  });

  it('should render breadcrumbs correctly', () => {
    const props = {
      ...baseProps,
      breadcrumbs: [{ path: '/blog', text: 'Blog' }],
    };
    const element = shallow(<Navbar {...props} />);

    const links = element.find(Link);
    const dividers = element.find('.divider');

    expect(links).toHaveLength(2);
    expect(dividers).toHaveLength(1);

    expect(links.at(0).props().children).toBe('Jack Lewin');
    expect(links.at(0).props().to).toBe('/');
    expect(links.at(1).props().children).toBe('Blog');
    expect(links.at(1).props().to).toBe('/blog');
  });
});
