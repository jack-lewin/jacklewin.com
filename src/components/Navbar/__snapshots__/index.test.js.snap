// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    breadcrumbs={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "main-navigation",
      "children": Array [
        <GatsbyLink
          to="/"
        >
          Jack Lewin
        </GatsbyLink>,
        Array [],
      ],
      "className": "container",
      "role": "navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Jack Lewin",
          "to": "/",
        },
        "ref": null,
        "rendered": "Jack Lewin",
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "main-navigation",
        "children": Array [
          <GatsbyLink
            to="/"
          >
            Jack Lewin
          </GatsbyLink>,
          Array [],
        ],
        "className": "container",
        "role": "navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Jack Lewin",
            "to": "/",
          },
          "ref": null,
          "rendered": "Jack Lewin",
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
